### DISCUSSION TITLE
Missing longitude and latitude values in GOES-16 FOV

### DISCUSSION THREAD

**Discussion by jwynnsmith** [blaylockbk/goes2go#18] [Q&A]
I hope you are well.
I am Jonathan Smith a Scientist at GFDL working on the development of the lightning frequency and LNOx parameterizations in climate models.

I recently used your script here: https://github.com/blaylockbk/goes2go/blob/master/latlon_grids/create_standard_GOES_latlon_grids.ipynb to generate the the GOES 16 FOV.  As you know it is 5424 x 5424 file and it has missing values  

I presume that the missing coordinate values in these files come from the projection of the GOES-16 FOV onto a sphere.  

With most of the regridding software (cdo, ncl ESMF Regrid, nco ncremap, etc), a netcdf file needs to include variables that contain the 4 corners of each grid cell. So I have added those variables.  In the case of the GOES FOV file, a subset of grids will either have all or 1-3 of the 4 are missing values.  The missing values are usually printed out as a large number like 1e+20 or 9.96e+36, hence the calculation is thrown off. And I usually try to do a conservative regridding since I am going from a very fine 5424 x 5424 to 128 (1.25 degree Lon) to 162 (1 degree lat). 

Would it be wise to do some type of interpolation to estimate those missing coordinates?  Or should I use a fixed coordinate system where the coordinates are (0,0) to (5424,5424)?

Thanks
Jonathan

**Comment by blaylockbk** [comment]
Hi @jwynnsmith 

I am familiar with the regridding problem you have run into. It's one I haven't managed to understand or solve myself yet. I have learned that transforming between coordinate systems and regridding is not easy 😄

There might be a solution somewhere in the Proj4 library that would require some digging. I have used Cartopy to transform from the geostationary coordinates to a different coordinate system (bottom of this [notebook](https://nbviewer.org/github/blaylockbk/goes2go/blob/master/docs/user_guide/notebooks/mapping_GOES16_TrueColor_Cartopy.ipynb)). I don't know that if gives you any clues. It might be wise to work in the geostationary projection coordinates to do the regridding, then transform the grid to lat/lon coordinates.

Another thing I have done was set all the NaN values to some lat/lon point in the Gulf of Alaska. I don't know how any regridding software would deal with that. It might be wise to work in the geostationary projection coordinates to do the regridding, then transform the grid to lat/lon coordinates after the regridding is done.

**Comment by jwynnsmith** [comment]
Thank you for your reply!! I can confirm this is a difficult venture but thank you for the tips.  I am familiar with converting x/y points to lat/lon using Proj4.  The setting to a GOA coordinate may be the way to go as well.  I am currently working with a xESMF and using an input source file created with ncl ESMF_Regrid to see that works.

**Reply by blaylockbk** [reply]
No problem. Let me know here when you come up with a solution. I'm curious how this can be done.
